core/coreunix/add.go

```diff
@@ -470,8 +471,14 @@ func (adder *Adder) addFile(file files.File) error {
 		return err
 	}
 
+	addFileName := file.FileName()
+	addFileInfo := file.(files.FileInfo)
```

Comment:
@keks are you *sure* this won't panic in some cases? We should probably use a checked type assertion (or comment on why this *must* be true).

Reply:
Hi @schomatis , i have add a checked type assertion in pr.Maybe it will be more robust.